### DISCUSSION TITLE
What CPU / RAM are recommanded to make the API to work properly

### DISCUSSION THREAD

**Discussion by stouch** [komoot/photon#726] [Q&A]
I suppose that we need to store the Elasticsearch data in a NVMe or SSD volume. 
What are the recommanded server performance in term of CPU / RAM ?

We've imported ~15 west europe countries, in 8 languages.

EDIT:
I've read this kind of issues https://github.com/komoot/photon/issues/258 so I guess we have to lower the Xmx parameter like  `-Xmx500m` but I'm not familiar with java app neither webservice configuration. How to setup the minimal heap size (-Xms) and the maximum (-Xmx) ? What is the incidence of CPU on the API response time ?

EDIT 2:
My instance has 15G RAM, 2vCores (2.2GHz), external SSD disk (2000 IOPS), and my queries are about 10sec slow, event when I set `-Xms6G -Xmx12G`
https://github.com/komoot/photon/issues/258
Should I change something ? 

EDIT 3:
I tested to improve the storage disk (from an external SSD 2K IOPS r/w, up to an internal ~30K), and it didn't change anything, queries are still slow, even when I re-imported on this disk only 2 countries instead of 15 (in 2 languages instead of 8). So it seems to be only a RAM / CPU issue. (Actually it improved delay from ~10sec to 3-5 sec for my test requests, but I would be expecting to get like 1-3 seconds for this self-managed service)

Thank you, 
Regards

**Comment by lonvia** [comment]
I don't have minimum specifications for hardware. I just can say what we run at https://photon.komoot.io for a planet-wide Photon installation. That's a i7-8700 with  6 CPUs, 64GB RAM and NVME SSD disks. Java runs in standard mode, i.e. no 'Xmx' parameters. That seems to work fairly well, see https://photon.komoot.io/munin/localdomain/localhost.localdomain/index.html for resource usage. More RAM is better for sure.

Run your test queries against the public instance to get a base line for query times. You should get well below 1s on average.

**Reply by stouch** [reply]
Ok I just made some tests with 30GB RAM, it's not much faster (still ~3 seconds) but ok I'll do more tests it might be the CPU too. I'll keep you in touch if I find a better configuration. Thanks.

**Comment by do-me** [comment]
Does anyone have more benchmarks for smaller instances, e.g. only EU, US, Asia or similar? For our use case we'd be interested in the minimum hardware specs needed for EU with very little traffic (maybe a hundred of requests per day).

**Reply by otbutz** [reply]
We have an EU instance running with `-Xmx2G`. Note that the import may require more than this.

**Reply by do-me** [reply]
That's pretty low, great! Thanks for sharing.
